src/Zynga/Framework/Cache/V2/Driver/InMemory.hh

```diff
@@ -0,0 +1,160 @@
+<?hh // strict
+
+namespace Zynga\Framework\Cache\V2\Driver;
+
+use Zynga\Framework\StorableObject\V1\Interfaces\StorableObjectInterface;
+use Zynga\Framework\Dynamic\V1\DynamicClassCreation;
+use Zynga\Framework\Cache\V2\Driver\Base as DriverBase;
+use Zynga\Framework\Cache\V2\Interfaces\DriverConfigInterface;
+use Zynga\Framework\Cache\V2\Interfaces\DriverInterface;
+use Zynga\Framework\Cache\V2\Exceptions\StorableObjectRequiredException;
+use Zynga\Framework\Cache\V2\Driver\InMemory as InMemoryDriver;
+use Zynga\Framework\Exception\V1\Exception;
+
+/**
+ * Use this class when you want to mock memcache.
+ */
+class InMemory extends DriverBase {
+  private static ?Map<string, mixed> $data;
+  private DriverConfigInterface $_config;
+
+  public function __construct(DriverConfigInterface $config) {
+    $this->_config = $config;
+  }
+
+  public function getConfig(): DriverConfigInterface {
+    return $this->_config;
+  }
+
+  public function getByMap(
+    Map<string, mixed> $data,
+  ): ?StorableObjectInterface {
+
+    try {
+
+      $className = $this->getConfig()->getStorableObjectName();
+
+      $obj = DynamicClassCreation::createClassByName($className, Vector {});
+
+      if (!$obj instanceof StorableObjectInterface) {
+        throw new StorableObjectRequiredException('className='.$className);
+      }
+
+      $obj->import()->fromMap($data);
```

Comment:
I think we need to revisit this. Given an in-memory caching structure, this is directly mapping the in memory cache to what the storable is. However, due note that the driver is intended to be a generalized cache. What is another way of doing this to support arbitrary structures?